kronecker.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import tensorflow as tf
+
+from tensor_train import TensorTrain
+
+def determinant(kron_a):
+  """Computes the determinant of a given matrix, factorized into
+  a Kronecker product of square matrices.
+  
+  Note, that this method can suffer from overflow.
+
+  Args:
+    kron_a: `TensorTrain` object containing a matrix of size N x N, 
+    factorized into a Kronecker product of square matrices (all 
+    tt-ranks are 1 and all tt-cores are square). 
+  
+  Returns:
+    Number, the determinant of the given matrix
+
+  Raises:
+    ValueError if the tt-cores of the provided matrix are not square,
+    or the tt-ranks are not 1
+  """
+  if not _is_kron(kron_a):
+    raise ValueError('The argument should be a Kronecker product (tt-ranks should be 1)')
+
+  cores = kron_a.tt_cores
+  det, pows = 1, 1
+  for core_idx in range(kron_a.ndims()):
+    core = kron_a.tt_cores[core_idx]
+    if core.get_shape()[1] != core.get_shape()[2]:
+      raise ValueError('The argument should be a Kronecker product of square matrices' 
+                      '(tt-cores must be square)')
+    pows *= core.get_shape()[1].value
+  for core_idx in range(kron_a.ndims()):
+    core = cores[core_idx]
+    core_det = tf.matrix_determinant(core[0, :, :, 0])
+    core_pow = pows / core.get_shape()[1].value
+
+    det *= tf.pow(core_det, core_pow)
+  return det
+
+
+def slog_determinant(kron_a):
+  """Computes the sign and (natural) logarithm of the determinant of 
+  the given matrix, factorized intoa Kronecker product of square matrices.
+
+  Args:
+    kron_a: `TensorTrain` object containing a matrix of size N x N, 
+    factorized into a Kronecker product of square matrices (all 
+    tt-ranks are 1 and all tt-cores are square).  
+  Returns:
+    Number, the log-determinant of the given matrix
+
+  Raises:
+    ValueError if the tt-cores of the provided matrix are not square,
+    or the tt-ranks are not 1
+  """
+  if not _is_kron(kron_a):
+    raise ValueError('The argument should be a Kronecker product (tt-ranks should be 1)')
+  
+  pows = 1
+  for core_idx in range(kron_a.ndims()):
+    core = kron_a.tt_cores[core_idx]
+    if core.get_shape()[1] != core.get_shape()[2]:
+      raise ValueError('The argument should be a Kronecker product of square matrices' 
+                      '(tt-cores must be square)')
+    pows *= core.get_shape()[1].value
```

Comment:
We don't need to compute the log of this product here, so I used what you suggested in the previous comment here